src/main/java/tech/zerofiltre/blog/infra/providers/database/course/LessonJPARepository.java

```diff
@@ -18,4 +18,10 @@ public interface LessonJPARepository extends JpaRepository<LessonJPA, Long> {
             "LOWER(l.content) LIKE LOWER(CONCAT('%', :keyword, '%')) OR " +
             "LOWER(l.summary) LIKE LOWER(CONCAT('%', :keyword, '%'))) ")
     List<LessonWithCourseIdJPA> findByKeyword(String keyword, Status status);
+
+    @Query(value = "SELECT l.id FROM lesson l " +
+            "JOIN chapter c ON c.id = l.chapter_id AND c.course_id = ?1 " +
```

Comment:
remplace : AND c.course_id = ?1 par

AND c.course_id = (SELECT courseId from enrollment erl where erl.id=?1)
... le reste est pareil